polaris/polaris/models.py

```diff
@@ -261,26 +238,64 @@ def __init__(self, *args, **kwargs):
     symbol = models.TextField(default="$")
     """The symbol used in HTML pages when displaying amounts of this asset"""
 
-    distribution_account_signers = models.TextField(null=True, blank=True)
-    """The JSON-serialized signers object returned from Horizon"""
-
-    distribution_account_thresholds = models.TextField(null=True, blank=True)
-    """The JSON-serialized thresholds object returned from Horizon"""
-
-    distribution_account_master_signer = models.TextField(null=True, blank=True)
-    """
-    The JSON-serialized object returned from Horizon for the object containing 
-    the account's public key, if present.
-    """
-
     objects = models.Manager()
 
     @property
     def distribution_account(self):
+        """
+        The Stellar public key derived from `Asset.distribution_seed`
+        """
         if not self.distribution_seed:
             return None
         return Keypair.from_secret(str(self.distribution_seed)).public_key
 
+    @property
+    def distribution_account_master_signer(self):
+        if self.distribution_account and not self._distribution_account_loaded():
+            self.load_distribution_account_data()
+        return self._distribution_account_master_signer
+
+    @property
+    def distribution_account_thresholds(self):
+        if self.distribution_account and not self._distribution_account_loaded():
+            self.load_distribution_account_data()
+        return self._distribution_account_thresholds
+
+    @property
+    def distribution_account_signers(self):
+        if self.distribution_account and not self._distribution_account_loaded():
+            self.load_distribution_account_data()
+        return self._distribution_account_signers
+
+    def _distribution_account_loaded(self):
+        return self.distribution_account and all(
+            f is not None
+            for f in [
+                self._distribution_account_master_signer,
+                self._distribution_account_signers,
+                self._distribution_account_thresholds,
+            ]
+        )
+
+    def load_distribution_account_data(self):
+        from polaris import settings
+
+        if self.code in ASSET_DISTRIBUTION_ACCOUNT_DATA:
+            account_json = ASSET_DISTRIBUTION_ACCOUNT_DATA[self.code]
+        else:
+            account_json = (
+                settings.HORIZON_SERVER.accounts()
```

Comment:
if possible I'd wrap a try-catch around this horizon API call in case it fails. In which case I think it would leave the private fields empty, so it could be loaded later